### PR TITLE
Update sft_trainer.py

### LINKED ISSUE #473: SFTTrainer: passing a PeftModel as a model argument without passing the config breaks
[https://github.com/lvwerra/trl/blob/main/trl/trainer/sft_trainer.py#L127C36-L127C47](https://github.com/lvwerra/trl/blob/main/trl/trainer/sft_trainer.py#L127C36-L127C47) Requires that you pass a PEFT config even though you've already instantiated the PeftModel yourself, otherwise it tries ignores the if and tries to build a regular transformers model at https://github.com/lvwerra/trl/blob/main/trl/trainer/sft_trainer.py#L148 taking your PeftModel as argument, which makes no sense.

I suggest adding a check at https://github.com/lvwerra/trl/blob/main/trl/trainer/sft_trainer.py#L147 to make sure that not isinstance(model, PeftModel):

```python
elif not isinstance(model, PreTrainedModel) and not isinstance(model, PeftModel):
```

### PR DESCRIPTION
**Pull request by JulesGM**
Allows the user to give their own peft model arg. https://github.com/lvwerra/trl/issues/473

Fixes #473

### CODE REVIEW

File: trl/trainer/sft_trainer.py [hunk 1]
```diff
@@ -144,7 +144,7 @@ def __init__(
 
             if callbacks is None:
                 callbacks = [PeftSavingCallback]
-        elif not isinstance(model, PreTrainedModel):
+        elif not isinstance(model, PreTrainedModel) and not isinstance(model, PeftModel):
```
Comment (younesbelkada) [inline trl/trainer/sft_trainer.py:147]:
```suggestion
        elif isinstance(model, str):
```

**Review by younesbelkada — COMMENTED** [review]
Great catch! Thanks for fixing, proposed an alternative, what do you think?

**Comment by JulesGM** [comment]
The goal of still using the SFTTrainer is to use the peft saving callback, and maybe the dataset saving stuff I guess

File: trl/trainer/sft_trainer.py (hunk 1 above)
Comment (JulesGM) [reply trl/trainer/sft_trainer.py:147]: what about

```python
elif not isinstance(model, (PretrainedModel, PeftModel):
```
which is the same thing but more compact I guess. 

`str` fails on `pathlib.Path` to a local save of a pretrained model, though I don't know if that's supported to begin with

Comment (JulesGM) [reply trl/trainer/sft_trainer.py:147]: but I'm fine with that too

Comment (younesbelkada) [reply trl/trainer/sft_trainer.py:147]: I see, can you change it with the suggesiton you made then ? 🙏 Then we can merge I think!

Comment (JulesGM) [reply trl/trainer/sft_trainer.py]: ok I pushed the suggestion

**Review by younesbelkada — APPROVED** [review]
Thank you for the clean work !

**Pull request merged**